### DISCUSSION TITLE
delete user from python supabase client

### DISCUSSION THREAD

**Discussion by patilrush9** [supabase/supabase#19222] [Contribute to Supabase]
delete user from python supabase client getting error AttributeError: 'SupabaseAuthClient' object has no attribute 'delete_user'

**Comment by 364499361** [comment]
Supabase's Python client does not directly provide the ability to delete users. This is because Supabase is a database-as-a-service (DBaaS), which does not directly manage user accounts, but focuses on processing and storing data. Deleting users is typically done in your application's account system, not in Supabase itself.

However, if you need to perform this operation in Supabase, you may need to use Supabase's API client. The Supabase API provides complete control over the database, including deleting records.

First, you need to install supabase-python-client, which is a Python client library for interacting with the Supabase API. You can install it using pip:

Bash
pip install supabase-python-client
Then, you can create a Supabase API client and use it to perform any desired operation. For example, if you want to delete a user, you may need to find the user's ID and then send a DELETE request to /users/{id}.

Here is an example code snippet:
from supabase import SupabaseClient, SupabaseException  
from supabase.api import APIClient  
import requests  
  
# Initialize Supabase client  
supabase = SupabaseClient(base_url='YOUR_SUPABASE_URL', key='YOUR_SUPABASE_KEY')  
  
# Initialize API client  
api_client = APIClient(supabase)  
  
# Find the user ID you want to delete  
user_id = 'THE_USER_ID_YOU_WANT_TO_DELETE'  
  
try:  
    # Delete the user  
    response = api_client.delete(f'/users/{user_id}')  
    print(response.text)  
except SupabaseException as e:  
    print(f'An error occurred: {e}')

**Reply by patilrush9** [reply]
getting error in installation 
ERROR: Could not find a version that satisfies the requirement supabase-python-client (from versions: none)
ERROR: No matching distribution found for supabase-python-client

**Reply by vatsal2473** [reply]
supabase_client.auth.admin.delete_user(user_id)

try this

**Reply by augcampos** [reply]
I try use the `supabase_client.auth.admin.delete_user(user_id)` but even using the service_role key I always get an exception

**Comment by patilrush9** [comment]
supabase AI also giving all wrong solutions

**Comment by jainsaundarya** [comment]
ERROR: Could not find a version that satisfies the requirement supabase-python-client (from versions: none)
ERROR: No matching distribution found for supabase-python-client

Not even able to install the client
Python version: 3.8.10
